server/log_operation_manager_test.go

```diff
@@ -200,22 +244,82 @@ func TestLogOperationManagerOperationLoopPassesIDs(t *testing.T) {
 
 	mockLogOp := NewMockLogOperation(ctrl)
 	infoMatcher := logOpInfoMatcher{50}
-	mockLogOp.EXPECT().ExecutePass(gomock.Any(), logID1, infoMatcher).Do(func(_ context.Context, _ int64, _ *LogOperationInfo) {
+	mockLogOp.EXPECT().ExecutePass(gomock.Any(), logID1, infoMatcher).DoAndReturn(func(_ context.Context, _ int64, _ *LogOperationInfo) (int, error) {
 		if atomic.AddInt64(&logCount, 1) == 2 {
 			cancel()
 		}
+		return 1, nil
 	})
-	mockLogOp.EXPECT().ExecutePass(gomock.Any(), logID2, infoMatcher).Do(func(_ context.Context, _ int64, _ *LogOperationInfo) {
+	mockLogOp.EXPECT().ExecutePass(gomock.Any(), logID2, infoMatcher).DoAndReturn(func(_ context.Context, _ int64, _ *LogOperationInfo) (int, error) {
 		if atomic.AddInt64(&logCount, 1) == 2 {
 			cancel()
 		}
+		return 0, nil
 	})
 
 	info := defaultLogOperationInfo(registry)
 	lom := NewLogOperationManager(info, mockLogOp)
 	lom.OperationLoop(ctx)
 }
 
+func TestLogOperationManagerOperationLoopExecutePassError(t *testing.T) {
```

Comment:
Is there a way to share some of this code with the other test function above? Seems like the only principal difference between them is `OperationSingle` vs `OperationLoop`?

Reply:
It looks like it should be possible to dedup a fair amount of this code. I'll do this as a follow-up PR.

Reply:
SGTM, will be happy to review it.